src/matter/IoBrokerNodeStorage.ts

```diff
@@ -45,66 +39,74 @@ export class IoBrokerNodeStorage implements Storage {
             return;
         }
 
-        // read all keys
+        // read all keys, storage entries always have a value, so we can use the states
         const states = await this.adapter.getStatesAsync(`${this.storageRootOid}.*`);
-        const len = `${this.adapter.namespace}.${this.storageRootOid}.`.length;
+        const namespaceLength = this.adapter.namespace.length + 1;
         for (const key in states) {
-            this.createdKeys[key] = true;
-            this.data[key.substring(len)] = fromJson(states[key].val as string);
+            this.existingObjectIds.add(key.substring(namespaceLength));
         }
 
         this.initialized = true;
     }
 
     async clearAll(): Promise<void> {
-        await this.adapter.delObjectAsync(this.storageRootOid, { recursive: true });
+        try {
+            await this.adapter.delObjectAsync(this.storageRootOid, { recursive: true });
+        } catch (error) {
+            this.adapter.log.error(`[STORAGE] Cannot clear all state: ${error}`);
```

Comment:
logging error object on purpose or is `error.message` enough? When being explicit maybe `error.stack`?

Reply:
This error is so seldom and can only occur if no permissions or object not exist. So it is definitely enough just to indicate `error.toString()` to understand, what happens

Reply:
i see the same. Adjusted